### DISCUSSION TITLE
Filter the filter values when filtering from other filters

### DISCUSSION THREAD

**Discussion by codemonkey800** [chanzuckerberg/napari-hub#518] [Ideas]
<img width="255" alt="Screen Shot 2022-05-31 at 1 48 53 PM" src="https://user-images.githubusercontent.com/2176050/171282047-788f17f7-d886-41d7-bb4a-d2265c0d2461.png">

This was an idea that originated from a question from Jun.  When filtering on another filter, it should also do an internal filtering for the values of the other filters to reduce the amount of data we present to the user.
 
One edge case to consider is if the filters get to a state where some of the filters do not have values, so we would need design to indicate the filter is empty or not usable.

**Comment by neuromusic** [comment]
I love this. Its quite consistent with the filter logic, which does an AND across filters, so this would limit the user to only be able to select valid combinations given their prior selection(s) and prevent them from filtering down to "unavailable"

Its also consistent with how many e-commerce sites support filtering

**Reply by neuromusic** [reply]
I assume this would intersect with _search_, too?

That is, if I have searched for something I will only see filters that are relevant to the current search results?

**Reply by codemonkey800** [reply]
> prevent them from filtering down to "unavailable"

This is the key value I saw in this feature too 💯 

> I assume this would intersect with search, too?

Ahh didn't think about that, but yes I think search should also be integrated here too since it also narrows the plugin results
